src/user/user.database.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable } from '@nestjs/common'
+import { InjectModel } from '@nestjs/mongoose'
+import { Model } from 'mongoose'
+
+import { USER_MODEL } from '../common/constants/database.constants'
+import { ConflictError, NotFoundError } from '../common/errors/errors'
+import { hash } from '../common/utils/crypto.utils'
+import { CreateUserDto } from './dto/create-user.dto'
+import { UpdateUserDto } from './dto/update-user.dto'
+import type { User } from './schemas/user.schema'
+
+@Injectable()
+export class UserDatabaseService {
+  constructor(@InjectModel(USER_MODEL) private userModel: Model<User>) {}
+
+  async createUser(createUserDto: CreateUserDto): Promise<User> {
+    const userDuplicate = await this.userModel.findOne({
```

Comment:
Everything in this method is a business logic so should be moved to the `UserService`

Reply:
Fixed